src/storage/tracked.rs

```diff
@@ -0,0 +1,327 @@
+use std::iter::Enumerate;
+use std::marker::PhantomData;
+use std::ops::{Deref, DerefMut};
+use std::slice;
+
+use hibitset::{BitSet, BitSetLike};
+
+use {Component, DenseVecStorage, Index, MaskedStorage, Storage, UnprotectedStorage};
+
+#[derive(Copy, Clone, Debug, PartialEq, PartialOrd)]
+pub enum Change {
+    #[doc(hidden)]
+    None,
+    Inserted,
+    Modified,
+    Removed,
+}
+
+impl Change {
+    fn add(&mut self, change: Change) {
+        let new = match (*self, change) {
+            (Change::None, new) => new,
+            (Change::Inserted, _) => Change::Inserted,
```

Comment:
What if it `Inserted` and them `Removed`?

Reply:
Sorry not pushed that yet. It would be `None`